Migration docs for v5 to v6 (#3909)
Shopify/polaris-react: documentation/guides/migrating-from-v5-to-v6.md

# Migrating from v5 to v6

Polaris v6.0.0 ([full release notes](https://github.com/Shopify/polaris-react/releases/tag/v6.0.0)) removes deprecated features in past releases. This file describes all code updates required to stay up to date.

> ## newDesignLanguage feature has been removed and is now the default
> 
> In v6 you no longer need to enable the newDesignLanguage through the features flag in the `AppProvider`.

Reviewer: ```suggestion
## New Design Language

In v5 we exposed a feature flag toggle to change between our old and new design languages. In v6 the new design is always enabled by default and our old design language has been removed. 
In v6 you no longer need to enable the newDesignLanguage through the features flag in the `AppProvider`.
```

```diff
<AppProvider
  i18n={{}}
- features={{newDesignLanguage: true}}
>
```

## Link is underlined by default

> When using the `Link` component it is now underlined by default to make sure that color is not the only way for users to percieve interactivity.

Reviewer: ```suggestion
The `Link` component it is now underlined by default to make sure that color is not the only way for users to percieve interactivity.
```

> When other factors help a user determine interactivity the underline can be removed with `removeUnderline` prop.

Reviewer 2: ```suggestion
When other factors help a user determine interactivity the underline can be removed with the `removeUnderline` prop.
```

```diff
- <Link url="https://help.shopify.com/">Orders</Link>
+ <Link url="https://help.shopify.com/" removeUnderline>Orders</Link>
```

> ## Remove the light prop from Tooltip

Reviewer 2: ```suggestion
## Removed the light prop from Tooltip
```
Might we want to be past tense on these updates?

> With the new changes `Tooltip` it now defaults to a light background. The property can safely be removed.

Reviewer: ```suggestion
In the new design language `Tooltip`s are always light. Thus the `light` prop no longer has any effect and has been be removed. Remove any usage of this prop.
```

```diff
<Tooltip
  content="Hello world"
- light
>Order #1001</Tooltip>
```

> ## Modals require a title property that can be hidden
> 
> Modal titles are required for screenreaders. There is a new option to hide the title.

Reviewer: ```suggestion
## Modal title prop is now required

To help enforce accessibility for screenreaders, `Modal`'s title prop is now required. You must add a title to all your modals. In the event that you do not want to display this title visually you can add the `titleHidden` prop to hide the title.
```

```diff
<Modal
  activator={activator}
  open={active}
  onClose={handleChange}
  primaryAction={{
    content: 'Add Instagram',
    onAction: handleChange,
  }}
  secondaryActions={[
    {
      content: 'Learn more',
      onAction: handleChange,
    },
  ]}
+  title="Reach more shoppers with Instagram product tags"
+  hiddenTitle
>
  <Modal.Section>
    <TextContainer>
      <p>
        Use Instagram posts to share your products with millions of
        people. Let shoppers buy from your store without leaving
        Instagram.
      </p>
    </TextContainer>
  </Modal.Section>
</Modal>
```

## Sheet requires an accessibility label

> Screen reader accessibility labels are required for screen readers. This new label does not render anything to the page.

Reviewer 2: ```suggestion
Accessibility labels are required for screen readers. This new label does not render anything to the page.
```

+ interface AnimationProps {
+   in?: boolean;
+ }
```

## Removed export for `NewDesignLanguageColor` and `Color`

We no longer support `NewDesignLanguageColor` and `Color` types. They were used to type our icon colors and have been replaced with ` IconProps['color']`.

```diff
- import {Color, NewDesignLanguageColor, Icon} from '@shopify/polaris';
+ import {Icon, IconProps} from '@shopify/polaris';

interface PartyIconProps {
- color: Color | NewDesignLanguageColor;
+ color: IconProps['color'];
}
```

## Replaced `BaseAction` type with `Action`

`BaseAction` has been replaced with the `Action` type.

```diff
- import {BaseAction} from '@shopify/polaris';
+ import {Action} from '@shopify/polaris';

export interface Props {
- headerAction?: BaseAction;
+ headerAction?: Action;
}
```

## Changed `frameOffset` prop to accept a string

This allows relative units for the frame offset. Allowing users to take into account for font-size and matching up breakpoints.

```diff
<AppProvider
  i18n={{}}
  theme={{
-   frameOffset: 60,
+   frameOffset: '60px',
  }}
>
```

## Replace `ariaPressed` prop with `pressed` in `Button` and `UnstyledButton`

```diff

> - <Button ariaPressed={true}>Lets go</Button>

Reviewer 2: ```suggestion
- <Button ariaPressed={true}>Let’s go</Button>
```

> + <Button pressed={true}>Lets go</Button>

Reviewer 2: ```suggestion
+ <Button pressed={true}>Let’s go</Button>
```

```

## Removed `Button`'s `stretchContent` prop

Consumers should combine the `fullWidth` and `textAlign="left"` props instead.

```diff
- <Button stretchContent>Lets go</Button>
+ <Button fullWidth textAlign="left">Lets go</Button>
```

## Removed `Popover`/`PopoverOverlay`'s `preventAutoFocus` prop

There was duplicated functionality and we have decided to replace `preventAutofocus` with `autofocusTarget="none"`.

```diff
<Popover
  active={popoverActive}
  activator={activator}
  onClose={togglePopoverActive}
- preventAutofocus
+ autofocusTarget="none"
>
  <p>Hello world</p>
</Popover>
```

> ## Removed `button-filled-disabled` SASS mixin

Reviewer: ```suggestion
## Removed `button-filled-disabled` SASS function
```

> This mixin has been removed as `Button` `disabled` property implements this background.

Reviewer: It seems strange to talk about "use a component" when talking about the lower-level foundations. Presumably people are using sass mixins because our components don't work for them (either because it lacks functionality or they're not using React)?

```diff
.IconButton {
  padding: 0.8rem;
- background: button-filled-disabled();
}
```

> ## Removed `plain-button-background` SASS mixin

Reviewer: ```suggestion
## Removed `plain-button-background` SASS function
```

> This mixin has been removed as `Button` `plain` property implements this background.

Reviewer: It seems strange to talk about "use a component" when talking about the lower-level foundations. Presumably people are using sass mixins because our components don't work for them (either because it lacks functionality or they're not using React)?

## AppProvider theming changes

### newDesignLanguage feature has been removed and is now the default

In v6 you no longer need to enable the newDesignLanguage through the features flag in the `AppProvider`.

```diff
<AppProvider
  i18n={{}}
- features={{newDesignLanguage: true}}
>
```

### Removed legacy topbar theming

With the new visual styles the TopBar theming has been replaced with the `ThemeProvider`. If you want to change the `TopBar` appearance you can ovveride the theme or the CSS variables.

```diff
<AppProvider
  i18n={{}}
  theme={{
    colors: {
-     topBar: {
-       color: 'rgb(33,43,54)',
-       backgroundLighter: 'rgb(244, 246, 248)',
-       backgroundDarker: 'rgb(223, 227, 232)',
-       background: 'rgb(255,255,255)',
-     },
+     surface: 'rgb(0, 0, 0)',
+     onSurface: 'rgb(33,43,54)',
    },
  }}
>
```

### Changed `frameOffset` prop to accept a string

This allows relative units for the frame offset. Allowing users to take into account for font-size and matching up breakpoints.

```diff
<AppProvider
  i18n={{}}
  theme={{
-   frameOffset: 60,
+   frameOffset: '60px',
  }}
>
```

## Exported type changes

> ### Removed exported type`AnimationProps`

Reviewer: ```suggestion
### Removed `AnimationProps`
```

`AnimationProps` is a simple type that was used once in `@shopify/polaris` and we have decided to remove it.

```diff
- import {AnimationProps} from '@shopify/polaris';

+ interface AnimationProps {
+   in?: boolean;
+ }
```

> ### Removed export for `NewDesignLanguageColor` and `Color`

Reviewer: ```suggestion
### Removed `NewDesignLanguageColor` and `Color`
```

> We no longer support `NewDesignLanguageColor` and `Color` types. They were used to type our icon colors and have been replaced with ` IconProps['color']`.

Reviewer: This feels like a kinda odd wording - it's not that people should use IconProps['color'] instead oof Color/NewDesignLanguageColor, it's that they shouldn't need to access these colors in the first place. [heres how a similiar concept was described previously](https://github.com/Shopify/polaris-react/blob/main/documentation/guides/migrating-from-v3-to-v4.md#linklikecomponent-)

```suggestion
The `NewDesignLanguageColor` and `Color` that describes the argument you pass into Icon's color prop. Yoou should use `IconProps['color']` instead of referencing these low level types.
```

```diff
- import {Color, NewDesignLanguageColor, Icon} from '@shopify/polaris';
+ import {Icon, IconProps} from '@shopify/polaris';

interface PartyIconProps {
- color: Color | NewDesignLanguageColor;
+ color: IconProps['color'];
}
```

### Replaced `BaseAction` type with `Action`

`BaseAction` has been replaced with the `Action` type.

```diff
- import {BaseAction} from '@shopify/polaris';
+ import {Action} from '@shopify/polaris';

export interface Props {
- headerAction?: BaseAction;
+ headerAction?: Action;
}
```

> ## Component API changes

Reviewer: These feel more important than removed exports. Put this section above "Exported type changes"?

```

### Removed `Button`'s `stretchContent` prop

Consumers should combine the `fullWidth` and `textAlign="left"` props instead.

```diff
- <Button stretchContent>Lets go</Button>
+ <Button fullWidth textAlign="left">Lets go</Button>
```

### Removed `Popover`/`PopoverOverlay`'s `preventAutoFocus` prop

There was duplicated functionality and we have decided to replace `preventAutofocus` with `autofocusTarget="none"`.

```diff
<Popover
  active={popoverActive}
  activator={activator}
  onClose={togglePopoverActive}
- preventAutofocus
+ autofocusTarget="none"
>
  <p>Hello world</p>
</Popover>
```

### Removed the light prop from Tooltip

With the new changes `Tooltip` it now defaults to a light background. The property can safely be removed.

```diff
<Tooltip
  content="Hello world"
- light
>Order #1001</Tooltip>
```

### Removed `plain` property in `Pagination`

With the new visual styles we have decided to remove the plain property from `Pagination`. This makes sure the Pagination appears interactive to users.

```diff
<Pagination
  hasPrevious
  hasNext
  onPrevious={() => handlePrevious()}
  onNext={() => handleNext()}
- plain={true}
/>
```

> ### Removed `button-filled-disabled` SASS mixin

Reviewer: Stick these under a new h2 "Sass API Changes" perhaps, as they're not really about components?

This mixin has been removed as `Button` `disabled` property implements this background.

```diff
.IconButton {

>   padding: 0.8rem;
> - background: button-filled-disabled();

Reviewer: That's not how you use a mixin.

```suggestion
  @include button-filled-disabled();
  padding: 0.8rem;
```

}
```

> ### Removed `plain-button-background` SASS mixin

Reviewer: ```suggestion
### Removed `plain-button-background` SASS function
```

> This mixin has been removed as `Button` `plain` property implements this background.

Reviewer: ```suggestion
This function has been removed as `Button` `plain` property implements this background.
```

Though it feels rubbish to talk "use the Button component" when we're describing the sass api - presumably people using the sass API aren't using our components for some.

```diff
.IconButton {

>   padding: 0.8rem;
> - background: plain-button-background();

Reviewer: That's noot how you use a mixin

```suggestion
- @include plain-button-background;
  padding: 0.8rem;
```

}
```

> ### Removed `text-emphasis-placeholder` SASS mixin

Reviewer: ```suggestion
### Removed `text-emphasis-placeholder` SASS function
```

- <Page title="Settings" separator>
+ <Page title="Settings">
  <Layout>
    <Layout.AnnotatedSection title="Store details">
      <p>Annotated section content</p>
    </Layout.AnnotatedSection>
  </Layout>
</Page>
```

### Removed `Popover`/`PopoverOverlay`'s `preventAutoFocus` prop

There was duplicated functionality and we have decided to replace `preventAutofocus` with `autofocusTarget="none"`.

```diff
<Popover
  active={popoverActive}
  activator={activator}
  onClose={togglePopoverActive}
- preventAutofocus
+ autofocusTarget="none"
>
  <p>Hello world</p>
</Popover>
```

### Removed the light prop from Tooltip

In the new design language `Tooltip`s are always light. Thus the `light` prop no longer has any effect and has been removed. Remove any usage of this prop.

```diff
<Tooltip
  content="Hello world"
- light
>Order #1001</Tooltip>
```

### Removed `plain` property in `Pagination`

With the new visual styles we have decided to remove the plain property from `Pagination`. This makes sure the Pagination appears interactive to users.

```diff
<Pagination
  hasPrevious
  hasNext
  onPrevious={() => handlePrevious()}
  onNext={() => handleNext()}
- plain={true}
/>
```

> ### Renamed the `additionalMetaData` property to `additionalMetadata` in `Header`

Author: ```suggestion
### Renamed the `additionalMetaData` property to `additionalMetadata` in `Page`
```

> This keeps the capitilisation consistent in the `Header` and `Title` components.

Author: ```suggestion
This keeps the capitilisation consistent across the `Header` and `Title` components in `Page`.
```

Polaris v6.0.0 ([full release notes](https://github.com/Shopify/polaris-react/releases/tag/v6.0.0)) removes deprecated features in past releases. This file describes all code updates required to stay up to date.

## New Design Language

> In v5 Polaris contained two design languages that could be toggled using the `newDesignLanguage` feature on `AppProvider`. We have removed our old viisual styles, leaving only the new.

Author: ```suggestion
In v5 Polaris contained two design languages that could be toggled using the `newDesignLanguage` feature on `AppProvider`. We have removed our old visual styles, leaving only the new.
```
